homeassistant/components/mqtt/fan.py

```diff
@@ -170,11 +220,54 @@ def _setup_from_config(self, config):
             "SPEED_HIGH": config[CONF_PAYLOAD_HIGH_SPEED],
             "SPEED_OFF": config[CONF_PAYLOAD_OFF_SPEED],
         }
+        self._feature_legacy_speeds = False
+        if not self._topic[CONF_SPEED_COMMAND_TOPIC] is None:
```

Comment:
Mark CONF_SPEED_COMMAND_TOPIC as deprecated in the schema and add a comment in the code that support will be removed after a quarter (2021.7), that will allow us to simplify the MQTT fan implementation then 👍

Reply:
I will take steps to add the comment

Reply:
I have marked all `speeds` related code and tests that should be removed after end of support.